dbfit-java/core/src/main/java/dbfit/util/DbParameterAccessor.java

```diff
@@ -78,14 +83,24 @@ public void bindTo(StatementExecution cs, int ind) throws SQLException{
         this.cs=cs;
         this.index=ind;    
         if (direction != INPUT){
-            cs.registerOutParameter(ind, getSqlType());
+            cs.registerOutParameter(ind, getSqlType(), direction == RETURN_VALUE);
         }
     }
 
     public void set(Object value) throws Exception {
         if (direction == OUTPUT|| direction == RETURN_VALUE)
             throw new UnsupportedOperationException("Trying to set value of output parameter "+name);
-        cs.setObject(index, value, sqlType, userDefinedTypeName);
+        TypeTransformer dbfitToJDBCTransformer = null;
+        Object newValue;
+        if (value != null) {
+            dbfitToJDBCTransformer = dbfitToJDBCTransformerFactory.getTransformer(value.getClass());
+        }
+        if (dbfitToJDBCTransformer != null) {
+            newValue = dbfitToJDBCTransformer.transform(value);
+        } else {
+            newValue = value;
```

Comment:
> One idea I've been wondering about: why do we need Object -> Factory -> transformer -> transform(). Can't we make a UniversalTransformer: transform(object) - to transformed value directly.

I was thinking a while back "why can't the factory object itself perform the transformation or return null of there's no transformer"?.